### DISCUSSION TITLE
342d:e481 supported?

### DISCUSSION THREAD

**Discussion by derekjc** [irfanjmdn/r65#6] [General]
From the RK website[1], I see that the device  `{"vendorId": "0x342d","productId": "0xe453"}` is for the wireless version. The wired version is  `{"vendorId": "0x342d", "productId": "0xe481"}`.

Does the firmware in this repo support the wireless version of the hardware with wireless connectivity not enabled(description says Wired-only)?

[1]: https://rkgamingstore.com/collections/65-keyboards/products/r65-wired-gaming-mechanical-keyboard "VIA drivers" under "Software" is a zip that contains json files for a number of RK keyboards.

**Comment by iamdanielv** [comment]
You are right, the link from the website says:
``` json
  "name": "RK-R65",
  "vendorId": "0x342d",
  "productId": "0xe481",
```

This code is only for the wired version, but it has a different product ID.
If you want to use that json from their website, you can change the productID in this repo to match. Then re-compile and flash your keyboard with the new PID.

Look at keyboard.json in this repo:
```json
 "usb": {
        "device_version": "0.1.0",
        "pid": "0xE453",
        "vid": "0x342D",
        "suspend_wakeup_delay": 1000
    },
```

and change pid to `0xe481`.


The other way would be to use the json file from this repository.

Hope that helps!

**Reply by derekjc** [reply]
Hi @iamdanielv, Thanks for the quick response. I'll try that.

**Reply by dressedinblack5** [reply]
Hi, guys I managed to review the firm that corresponds to 342D:E480 wired version device, visit my repo
